### ISSUE TITLE
[$400] Integrate updated UI

### ISSUE THREAD

**Issue by jmgasper** [topcoder-platform/evernym-tc-wallet#33]
Now we want to change navigation a bit:

* In the left menu, we'll add `Wallet` and `Topcoder`
* On initial login we'll go right into the app - initial view should just be a basic tutorial
* When the user selects `Wallet`, *then* we will initialise the wallet, if not already done
* When the user selects `Topcoder` they will get additional options for `Login` and `Challenges`
  * When the user selects `Login`, we'll log them in on the TC site
  * When the user selects `Challenges` and aren't logged in, they'll see all challenges 
    *  If we can design a similar filter mechanism to the community app, that would help
  * When the user selects `Challenges` and they're logged in, they'll see their current challenges

Design is updated:

- Marvel Link: https://marvelapp.com/prototype/5ig7b8c

- Storyboard Updates: https://drive.google.com/drive/folders/1HJXDbA8vG5c5Qp7c11gpDGJ7bMr3lXIk?usp=sharing

### Notes

For the challenge screens, we can assume that we'll just load the site in a web view for now - no need to try to mimic the UI natively.

**Comment by jmgasper** [comment]
Challenge https://www.topcoder.com/challenges/3dd462cf-c934-4568-8a08-39f0e51a14a1 has been created for this ticket.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by jmgasper** [comment]
Challenge https://www.topcoder.com/challenges/3dd462cf-c934-4568-8a08-39f0e51a14a1 has been assigned to seriyvolk83.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by seriyvolk83** [comment]
In progress. I will complete it tomorrow.

**Comment by jmgasper** [comment]
Great, thanks for the update.

**Comment by seriyvolk83** [comment]
Challenges page is shown by opening https://www.topcoder.com/challenges in embedded web view.

>If we can design a similar filter mechanism to the community app, that would help

I didn't implement it yet. If you want it to be controls in native app that will update the web content, then I need some details how to apply filter by modifying URL (GET parameters).


Empty Home page in updated design differs from what we have in the app. Should I update it too? What to do when user taps on one of the tutorial buttons, e.g. "Topcoder Wallet Tutorial"?

**Comment by seriyvolk83** [comment]
Tutorial after login is shown every time user logs in (or opens the app after it's suspended). Should it be shown only once (when completed, it will not appear any more)?

**Comment by jmgasper** [comment]
@seriyvolk83 - One question here:

Can we use the Auth0 login to also login in the web view?  So, if I login using the `Topcoder` option, can I *also* be logged in in the web view?  That'll keep us from having redundant logins.

**Comment by seriyvolk83** [comment]
Seems it's possible only if we open `https://accounts-auth0.topcoder.com/#!/member?retUrl=http://www.topcoder.com/` in embed web view. But, in this case the process of authentication is out of control. We cannot check if user logged in or not, the tokens are unavailable.

I tried to use "Lock" framework that allows to embed Login form, but it just native UI that does the same what Safari login form does - provides tokens to the client (callbacks with credentials). But, in such case we cannot "put" the credentials into a random web view.

There is something about cookie authentication. I think there is a way to put prepared cookies into a web view, but I don't know how to get that cookies from credentials (if possible). https://auth0.com/docs/sessions-and-cookies/authentication-api-cookies
The names of the cookies do not match cookies used in topcoder.com.

**Comment by seriyvolk83** [comment]
Try the app with the web view (last PR).

On other side, we can use credentials to call Auth0 API, and seems we can use them to call topcoder API to pull the content, but it's another story where content should be rendered natively.

**Comment by jmgasper** [comment]
Worked with the latest PR.

**Comment by jmgasper** [comment]
Payment task has been updated: https://www.topcoder.com/challenges/3dd462cf-c934-4568-8a08-39f0e51a14a1
*Payments Complete*
Winner: seriyvolk83
Copilot: ghostar
Challenge `3dd462cf-c934-4568-8a08-39f0e51a14a1` has been paid and closed.<br/><br/>```This is an automated message for ghostar via Topcoder X```
